### ISSUE TITLE
No Capacities assigned or found

### ISSUE THREAD

**Issue by RabidCadaver** [NielZeeman/team-capacity-extension#10]
Okay, I have the extension installed in an on premise install of TFS 2015.2

I've made sure that I have capacity specified for team members, and that tasks assigned to them are properly identified and estimated. When I click into the tab for the team capacity, it displays the titular error message "No capacities assigned or found"

**Comment by NielZeeman** [comment]
Hi,
Can you please

1. check what the default iterations are set to in the team configurations?
2. provide me the console log from the browser when you load the extension?

**Comment by RabidCadaver** [comment]
Presuming you mean the backlog iteration, it's set at the root of the other iterations. It also has dates specified on it. I think you've hit the nail on the head considering the last line found in the console. 

Creating new TelemetryClient!
index.js:36 TelemetryClient settings key: 69884dc6************
index.js:36 TelemetryClient settings extension context: TeamCapacity
index.js:36 TelemetryClient settings disableTelemetry: false
index.js:36 TelemetryClient settings disableAjaxTracking: false
index.js:36 TelemetryClient settings enableDebug: true
index.js:20 WorkService.loadTeam.findTeamSettings : No default iteration found

**Comment by RabidCadaver** [comment]
Which is perhaps because we're using the default team for the team project that we're in. Thus I don't have iterations for the default team to be able to set. Making a new one, I could specify iterations.

**Comment by NielZeeman** [comment]
Let me know if that works. 
At this stage it uses the current "system" default iteration which is derived from the team's selected iterations and the dates associated to them, as long as the current date falls _within_ a set of dates on one of the selected iterations.
It is on the backlog to have the iteration selectable to overcome this issue.

**Comment by RabidCadaver** [comment]
Still the same error. "WorkService.loadTeam.findTeamSettings : No default iteration found"

I defined a new team, stuck myself in it, and then made a new iteration, some sub iterations, assigned the backlog to that specific iteration root. Then I made a new requirement, some subtasks, made sure the tasks were in the current sub iteration, parsed out estimates and so on. The capacity page, where I've also assigned capacities matching the tasks. The details show up and everything.

**Comment by NielZeeman** [comment]
Does the iteration root have a start date in the past and a end date in the future?

**Comment by RabidCadaver** [comment]
Yessir. 
![3-8-2018 10-14-44 am](https://user-images.githubusercontent.com/3519375/37161952-f1895b0e-22eb-11e8-813b-6573b158796e.png)

**Comment by NielZeeman** [comment]
It seems that TFS 2015 does not report the default iteration.
Based on the [documentation ](https://www.visualstudio.com/en-us/docs/integrate/api/work/team-settings) should return a defaultIteration when you run something like this: https://{instance}/DefaultCollection/{project}/{team}/_apis/Work/TeamSettings

Even though TFS 2015 [should ](https://docs.microsoft.com/en-us/vsts/integrate/concepts/rest-api-versioning) support API version 2.0, I am getting contradicting results.

I've just tested it, and it does not seem to return the default iteration at all.

**Comment by RabidCadaver** [comment]
Yeah. The JSON it spit at me has a backlog iteration listed, but not a default one.

**Comment by RabidCadaver** [comment]
``

**Comment by RabidCadaver** [comment]
`
[json.txt](https://github.com/NielZeeman/team-capacity-extension/files/1794545/json.txt)
`

**Comment by mailravit** [comment]
I hit the same problem with our onprem TFS2015.4 server.    Hope we have some fix sooner.  We are really needed this functionality.   Hope will have  fix sooner.

**Comment by MTRH** [comment]
I also get the same message ("No Capacities assigned or found") after  installing the extension. I can see that the capacity has been entered.
How can I get this to work?

**Comment by NielZeeman** [comment]
I have completed the work to supply a work around, but am blocked by another issue: https://github.com/OfficeDev/office-ui-fabric-react/issues/4329
Waiting for feedback before I release a new version.

**Comment by samutrej** [comment]
Any update on when you might release a new version that includes the fix above? We seem to have the same issue with "default iteration not found" . I have checked and played around with that but have not been able to get the extension to work.
